processing/src/main/java/io/druid/segment/TimeAndDimsPointer.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment;
+
+import com.google.common.base.Preconditions;
+import io.druid.java.util.common.DateTimes;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * TimeAndDimsPointer is used in conjunction with {@link TimeAndDimsIterator}, it's an _immutable_ object that points to
+ * different logical data points, as {@link TimeAndDimsIterator#moveToNext()} is called.
+ *
+ * TimeAndDimsPointers are comparable by time and dimension column values, but excluding metric column values, to
+ * facilicate ordering and rollup during merging of collections of rows (see {@link IndexMergerV9#merge} methods).
+ *
+ * The difference between "time and dims" and "row" abstraction (see {@link
+ * io.druid.segment.incremental.IncrementalIndexRow}, {@link RowPointer}) is that "time and dims" is logical composite
+ * of only time point and dimension and metric values, not tied to a specific position in any data structure (aka "row
+ * index").
+ *
+ * TimeAndDimsPointer intrface is conceptually similar to {@link Cursor}, that is used for query execution rather than
```

Comment:
nit, typo 'intrface' (but seriously, :+1: on all these javadocs and comments)

Reply:
Thanks. Additionally, TimeAndDimsPointer is not an interface, it's a class. Fixed comments.